web/source/osk/visualKeyboard.ts

```diff
@@ -1119,16 +1120,36 @@ namespace com.keyman.osk {
     showLanguage() {
       let keyman = com.keyman.singleton;
 
-      var lgName='',kbdName='';
-      var activeStub = keyman.keyboardManager.activeStub;
+      let displayName: string = undefined;
+      let activeStub = keyman.keyboardManager.activeStub;
 
       if(activeStub) {
-        lgName=activeStub['KL'];
-        kbdName=activeStub['KN'];
-      } else if(keyman.getActiveLanguage(true)) {
-        lgName=keyman.getActiveLanguage(true);
+        if(activeStub['displayName']) {
+          displayName = activeStub['displayName'];
+        } else {
+          let
+            lgName: string = activeStub['KL'],
+            kbdName: string = activeStub['KN'];
+          kbdName = kbdName.replace(/\s*keyboard\s*/i,'');
+          switch(keyman.options['spacebarText']) {
+            case SpacebarText.KEYBOARD:
+              displayName = kbdName;
+              break;
+            case SpacebarText.LANGUAGE:
+              displayName = lgName;
+              break;
+            case SpacebarText.LANGUAGE_KEYBOARD:
+              displayName = (kbdName == lgName) ? lgName : lgName + ' - ' + kbdName;
```

Comment:
Not the old

```suggestion
              displayName = (kbdName == lgName) ? lgName : lgName+' ('+kbdName+')';
```

?

Reply:
As I wrote:

> * Sets the display pattern for LANGUAGE_KEYBOARD to be "language - keyboard" instead of "language (keyboard)", which reduces the number of parentheses visible on the spacebar.

See also https://github.com/keymanapp/keyman/issues/949#issuecomment-867169576

Reply:
From an alternate channel:

> Doesn’t seem that difficult to understand to me.  Khmer, Central (Khmer) (Khmer Angkor) or English (Latin) (EuroLatin (SIL)) is pretty messy. Khmer, Central (Khmer) - Khmer Angkor and English (Latin) - EuroLatin (SIL) are slightly better.

Unfortunately, that wasn't in the referenced comment.  The examples stated in this quote provided the context I was missing.